### PR TITLE
Adding bank dump for vbnt-1

### PR DESCRIPTION
**Pull request by nutterthanos**
hello i would like to add a bank dump for the TG789V3/vbnt-1
thanks nutterthanos

### CODE REVIEW

**Comment by kevdagoat** [comment]
Ok will see to merging

Sent from Mail for Windows 10

From: nutterthanos
Sent: Sunday, June 16, 2019 2:17 PM
To: kevdagoat/hack-technicolor
Cc: Subscribed
Subject: [kevdagoat/hack-technicolor] Adding bank dump for vbnt-1 (#5)

hello i would like to add a bank dump for the TG789V3/vbnt-1
thanks nutterthanos

You can view, comment on, or merge this pull request online at:
  https://github.com/kevdagoat/hack-technicolor/pull/5
Commit Summary
• bank dump for: vbnt-1_18.3.0157-3-2-1-CRF905
• Add Firmware dump link for vbnt-1/TG789V3
File Changes
• M docs/Firmware Repository.md (10) 
• A firmware/vbnt-1_18.3.0157-3-2-1-CRF905-bank_dump.xz (0) 
Patch Links:
• https://github.com/kevdagoat/hack-technicolor/pull/5.patch
• https://github.com/kevdagoat/hack-technicolor/pull/5.diff

**Comment by LuKePicci** [comment]
Please consider sharing its OSCK too!

**Comment by kevdagoat** [comment]
@LukePicci Good point

Sent from Mail for Windows 10

**Comment by nutterthanos** [comment]
How do we get the OSCK?

**Comment by LuKePicci** [comment]
TL;DR there is no easy way to do it on 18.x firmwares at the moment, just make the eripv2 dump and wait for some updated tools to be available

Generally speaking, you need to dump the eripv2 partition (mtd5), then unpack it with some tools available here: https://github.com/pedro-n-rocha/secr

That unpacking tool require you ro provide both the eripv2 dump and the ECK, which can be found in ram. For older firmwares with older kernels, there are kmods which print out that ECK. Where no compatible kmod is available, as in your case, you could search for it into a full ram dump made with some tool like LIME, which you need to build for your platform.

**Comment by Ansuel** [comment]
@LuKePicci actually on new platfrom the eck position should be passed by the bootline arg via the tag

platform.prozone_addr=0x1ffe0000

**Comment by Ansuel** [comment]
the key should be this

dd if=/dev/mem bs=1 count=16 skip=$eck_position | hexdump -e '16/1 "%02x " "\n"' |  sed 's/ //g'  and replace eck_position with the address in prozone

**Comment by LuKePicci** [comment]
I knew about the prozone and r2secr in bootline but I never had chance or time to check a memdump to see what was there. Thanks!

**Comment by kevdagoat** [comment]
@Ansuel Interesting. Looks like secr needs to be adapted

**Comment by LuKePicci** [comment]
Ok, I confirm, that r2secr address is what r2secr module look for, but it takes into account some extra memory mapping of the tch rip driver. In my case, on 3.4 arm, the ECK is 32 byte after the .r2secr base address.

So:
```bash
let eck_position=xxxxxxxxx+32
dd if=/dev/mem bs=1 count=16 skip=$eck_position | hexdump -e '16/1 "%02X " "\n"'
```

**Comment by Ansuel** [comment]
think it's standard...  the first 32 should be the EIK followed by the ECK

**Comment by LuKePicci** [comment]
I don't think so, I have one 4 byte address at +8 and some other sparse values. EIK must be an asymmetric key, being an integrity key, and as such could not fit in 32 bytes - it would be quite weak.

**Comment by arthurcanon** [comment]
On 17.1.X firmware (Kernel version 4.1.38) on VANT-9 and VBNT-Z, The string 'prozone' does not appear in the ttyS0 bootlog at all, so I cannot get the memory offset from there.

This is what I see in the bootlog:
---- VBNT-Z ----
[    0.000000] Kernel command line:  memsize=0xFFDD000 btab=0xc004080c btab_bootid=1 tbbt_addr=0x7d20000 board=VBNT-Z console=ttyS0,115200 root=/dev/mtdblock1 rootfstype=squashfs irqaffinity=0 console=ttyS0,115200 root=/dev/mtdblock1 rootfstype=squashfs irqaffinity=0 console=ttyS0,115200 root=/dev/mtdblock1 rootfstype=squashfs irqaffinity=0

---- VANT-9 ----
[    0.000000] Kernel command line: root=31:0 ro noinitrd memsize=0xFFDD000 btab=0xc004180c btab_bootid=2 tbbt_addr=0x7d20000 board=VANT-9 console=ttyS0,115200 root=/dev/mtdblock1 rootfstype=squashfs irqaffinity=0 console=ttyS0,115200 root=/dev/mtdblock1 rootfstype=squashfs irqaffinity=0

**Comment by arthurcanon** [comment]
Given that we have a memory dump, and a dump of the eripv2 partition (mtd5), would it be possible to conduct a known-plaintext attack, by iterating through the memory dump to test all possible keys? 256MB RAM = ~256 million possible keys, which should be fast on a GPU, or even a CPU.

Is there known-plaintext in the eripv2 partition?

**Comment by LuKePicci** [comment]
It is possible, when you test the right key To decrypt some tag, you can simply check eik signature to determine if that key was correct.

For example you can simply pass every single possible 16byte sequence you havenon the memory to eripv2.py until it prints “this proves eck is correct” (check the code for the exact string).

However it’s still easier to just read through the dump and search for known plaintext from any rip tags, for example I see an header certificate near the memory region where eck is found.

Da: arthurcanon
Inviato: giovedì 11 luglio 2019 04:42
A: kevdagoat/hack-technicolor
Cc: Luca Piccirillo; Mention
Oggetto: Re: [kevdagoat/hack-technicolor] Adding bank dump for vbnt-1 (#5)

Given that we have a memory dump, and a dump of the eripv2 partition (mtd5), would it be possible to conduct a known-plaintext attack, by iterating through the memory dump to test all possible keys? 256MB RAM = ~256 million possible keys, which should be fast on a GPU, or even a CPU.
Is there known-plaintext in the eripv2 partition?

**Comment by pedro-n-rocha** [comment]
@Ansuel , you once asked how TIM OSCK was retreived .. check rev1900 from my repos , i am letting it go ...

**Comment by Ansuel** [comment]
You mean modified driver? Can you link?

Il Ven 12 Lug 2019, 15:52 procha <notifications@github.com> ha scritto:

**Comment by nutterthanos** [comment]
OSCK has been shared
